testutil/loopback_linux.go

```diff
@@ -49,5 +69,47 @@ func NewLoopback(size int64) (string, func() error, error) {
 		return os.Remove(file.Name())
 	}
 
-	return deviceName, cleanup, nil
+	l := Loopback{
+		File: file.Name(),
+		Device: deviceName,
+		close: cleanup,
+	}
+	return &l, nil
+}
+
+// Loopback device
+type Loopback struct {
+	// File is the underlying sparse file
+	File string
+	// Device is /dev/loopX
+	Device string
+	close func() error
+}
+
+// SoftSize returns st_size
+func (l *Loopback) SoftSize() (int64, error) {
+	st, err := os.Stat(l.File)
+	if err != nil {
+		return 0, err
+	}
+	return st.Size(), nil
+}
+
+// HardSize returns st_blocks * 512
```

Comment:
Can you link to somewhere that justifies 512 as a hardcoded value

Reply:
http://man7.org/linux/man-pages/man2/fstat.2.html

> st_blocks
>  This field indicates the number of blocks allocated to the file, in 512-byte units.